legacy/firmware/fsm_msg_coin.h

```diff
@@ -144,6 +152,12 @@ void fsm_msgGetPublicKey(const GetPublicKey *msg) {
   resp->has_root_fingerprint = true;
   resp->root_fingerprint = root_fingerprint;
 
+  if (aborted) {
+    fsm_sendFailure(FailureType_Failure_ActionCancelled, NULL);
```

Comment:
any reason this is outside the while-loop?

Reply:
Sorry, my mistake... :facepalm: 
Fixed in 4b0f53ae9c.